src/Umbraco.Core/Persistence/Repositories/Implement/DocumentRepository.cs

```diff
@@ -468,6 +484,8 @@ protected override void PersistNewItem(IContent entity)
                 ClearEntityTags(entity, _tagRepository);
             }
 
+            PersistRelations(entity);
```

Comment:
I don't think we should be using an event at all since we aren't doing that anywhere else with the repositories in the code base. I've decided to explicitly call this base method from each inherited repository just to make it explicit, however, we could easily add this to the base classes methods of `PersistUpdatedItem` and `PersistNewItem` to have this automatically taken care of. I just thought for now to make it explicit but we can change if we like.